### DISCUSSION TITLE
Variable substitution for GitHub Template Repository Usage

### DISCUSSION THREAD

**Discussion by liach** [community/community#23378] [New to GitHub]
<p>GitHub template repositories often have contents that refer to the name of the template repo itself or the account that owns the repo (e.g. Travis-CI GitHub Page auto-publiscation service). I wonder if GitHub template repositories have such token substitution mechanism that substitutes certain elements in a template repo with correct repo owners/repo names.</p>

**Comment by logankilpatrick** [comment, marked as answer]
<p>Hey! As far as what I have read, this is not possible. It would be awesome to see this get added. I think the best thing we can do at this point is to add a feature request. </p>

**Comment by brianary** [comment]
<p>Where would we go to make or upvote this feature request?</p>

**Comment by joebowbeer** [comment]
Also:

https://github.com/orgs/community/discussions/5336

> A replacement for the organization would be a good start.

**Reply by da-amit-mujawar** [reply]
Use cookiecutter and cruft for this as of now

**Reply by ATorres-FresnelBio** [reply]
I have been using [Copier](https://copier.readthedocs.io/en/stable/), rather than cookiecutter. 
Copier has a story around [incorporating template updates](https://copier.readthedocs.io/en/stable/updating/) into instances of the template, using Git's commit history.

**Comment by dazinator** [comment]
Does anyone know if there has been any movement on this?
